### ISSUE TITLE
Cannot publish binary for Windows on webpack:webpack

### ISSUE THREAD

**Issue by eXon** [meteor/meteor#7661]
# Problem
I've made a new version of webpack:webpack and now, I need to publish the binaries. It all work on Mac/Linux. However, on Windows, I get this error on my own Windows computer (also tried on the VM):

# Reproduce
**Typing:** meteor publish-for-arch webpack:webpack@1.3.4

**Result:**

C:\Users\Admin>meteor publish-for-arch webpack:webpack@1.3.4
webpack:webpack: updating npm dependencies -- underscore, connect, cors,
webpack, webpack-dev-middleware, webpack-hot-middleware, memory-fs, mime...
C:\Users\Admin\AppData\Local\.meteor\packages\meteor-tool\1.4.0-1\mt-os.windows.
x86_32\isopackets\ddp\npm\node_modules\meteor\promise\node_modules\meteor-promis
e\promise_server.js:165
      throw error;
      ^

Error: ENOENT: no such file or directory, mkdir 'C:\Users\Admin\AppData\Local\Te
mp\package-source1d6frao\webpack_webpack\.npm\plugin\webpack:webpack-new-1sekc2k
'
    at Error (native)

# Solution
My guess is it is a problem because on Windows the folder webpack:webpack is webpack_webpack (can't use semicolons). This folder is generated by Meteor so I cannot change that.

Because of this, nobody on Windows can use my last version with a few major fixes so it's a pretty major bug for me.

**Comment by benjamn** [comment]
If you can wait for Meteor 1.4.1 (out tomorrow), `meteor publish-for-arch` will no longer be necessary. In fact, headaches like this one are what led us to [make it unnecessary](https://github.com/meteor/meteor/pull/7608).

That said, I think I can fix this problem directly. Thanks for bringing it to my attention!

**Comment by eXon** [comment]
@benjamn thanks for the fast answer! :) I can wait tomorrow and close this once it's working with the last version

**Comment by benjamn** [comment]
Ok, try `meteor update --release 1.4.1-rc.7`? Looking to publish the final 1.4.1 today, so please do let me know how it goes when you get a chance!

**Comment by benjamn** [comment]
Ok, I reproduced the problem with `1.4.1-rc.6` and verified that it's fixed in `1.4.1-rc.7`, so (especially given that `meteor publish-for-arch` is going away soon) I'm fairly confident we can close this. Of course feel free to reopen, if you believe otherwise!

**Comment by abernix** [comment]
@benjamn The change to correct this issue (61c1ab0cac9f437f7472d526cf391ea565c9a0bb + 4c1fe8ff2205922e0659a69b81cb5c5ddd139fe2) is causing the results seen in #7711.  There is a simple reproduction script there, but I also ran into it myself when trying to reproduce #7682 (i.e. cloning `tap:i18n` into packages/tap:i18).

As a test, I was able to use [this commit](https://github.com/abernix/meteor/commit/991eb1dd5ad0f534f533590b43b90e183cdac09d) to fix #7711.  To further evaluate this fix, I reproduced the problem above by trying to `publish-for-arch` my own version of `abernix:webpack` (a complete copy of `webpack:webpack` at the above version) using Meteor `1.4.1-rc.6`.  After applying this same fix, I could no longer reproduce the `ENOENT: no such file or directory, mkdir` error above.

I believe this would be a suitable alternative to the commits I mentioned in the first paragraph and fix #7711.  I see the importance of the colon converter method for portability, but applying it to anything which contains `sourceRoot` is a problem since `npmDir` becomes `npmCacheDirectory` and will be passed to `renameAlmostAtomically` and ultimately fail as the path on disk will be `packages/namespace:name/.npm` instead of `packages/namespace_name/.npm` (see #7711 for more on the actual fix).

The only entry-point I can see for this `npmCacheDirectory` being set (and thus `npmDir`) is from the call to `buildJsImage ` on [these lines](https://github.com/abernix/meteor/blob/991eb1dd5ad0f534f533590b43b90e183cdac09d/tools/isobuild/compiler.js#L81-L84) which seems to be properly colon-converting the package-name portion and leaving the `sourceRoot` as the path from the file system.

tl;dr Let me know your thoughts on using https://github.com/abernix/meteor/commit/991eb1dd5ad0f534f533590b43b90e183cdac09d in lieu of (61c1ab0cac9f437f7472d526cf391ea565c9a0bb + 4c1fe8ff2205922e0659a69b81cb5c5ddd139fe2)

**Comment by hwillson** [comment]
Hi all - just following up on this; the last comment in this thread (https://github.com/meteor/meteor/issues/7661#issuecomment-263936772) mentioned there might be an outstanding issue with the fix applied for this issue. While this issue is closed, it looks like it's related to the still open issue #7711 (where the fix applied here is causing that issue). Should we plan to move ahead with the fix outlined by @abernix in https://github.com/meteor/meteor/issues/7661#issuecomment-263936772? Let me know if I can help - thanks!
